pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -177,6 +183,21 @@ func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind
 	return shared.Informer().GetController(), shared.Informer().GetStore(), nil
 }
 
+// filterStoppedResources filters out the resources that have been removed
+// and reinstalled from the cluster in between garbage collector syncs.
+func (gb *GraphBuilder) filterStoppedResources(set resourceSet) {
+	gb.stoppedResourcesLock.Lock()
+	defer gb.stoppedResourcesLock.Unlock()
+	for resource := range gb.stoppedResources {
+		// filter resource from the resource set used by Sync
+		// so that it's deletion is recognized on the first Sync cycle.
+		delete(set, resource)
+		// filter the resource from gb's recentlyRemoved set
+		// so that it gets re-installed on the next sync cycle.
+		delete(gb.stoppedResources, resource)
```

Comment:
I'm not sure what you mean, can you explain?

Once the resource is removed from the `newResources` set (on line 194) it should be deleted from the `stoppedResources` set so that on the next iteration of Sync `stoppedResources` is ONLY populated with resources that have been stopped since the last time this filter was called (on the previous sync). 

Let me know if there's a better way to structure this code.

Reply:
1. so currently, the stopped resource needs to wait for one more sync cycle to get restarted, is it better to avoid the wait? it means gc.resyncMonitors(...) needs to be updated to understand the stopped & reinstalled resources in between syncs. it may be not worthwhile if the change is too complicated though.
2. "newResources = GetDeletableResources(discoveryClient)" is also called in the loop below the first call (at line 212 in garbagecollector.go), is filterStoppedResources(...) also needed there?
3. i was thinking to reset gb.stoppedResources outside the loop, but well, it works as is anyway.

Reply:
1. We only know about newly installed resources when gc.Sync runs and checks discovery (via GetDeletableResources), so I'm not sure there's a straightforward way to restart reinstalled resources between syncs, but something, somwhere still needs to poll discovery.
2. Good catch, added there as well. Ideally, I think the filtering would be a part of GetDeletableResources, but that is not a method on the gc and it’s probably not worth modifying the interface right?
3. Ah I see, changed.